.travis.yml

```diff
@@ -0,0 +1,34 @@
+sudo: required
+language: java
+
+env:
+  global:
+    - "MYSQL_DATABASE=irida_integration_test"
+    - "MYSQL_USER=test"
+    - "MYSQL_PASSWORD=test"
+  matrix:
+    - "TEST_SUITE=service_testing"
+    - "TEST_SUITE=rest_testing"
+    - "TEST_SUITE=galaxy_testing"
+    - "TEST_SUITE=ui_testing"
+
+services:
+  - docker
+
+addons:
+  mariadb: '10.0'
+
+jdk:
+  - openjdk8
+
+cache:
+  directories:
+    - $HOME/.m2
+    - src/main/webapp/node_modules/
```

Comment:
Is this cache maintained between test runs?  Or just for different phases of the same test?  If it's maintained between runs I'm thinking this should be removed from the cache.  This directory often needs to be blown away when there's changes to library versions.

Reply:
It's maintained between test runs.  If you have access to the travis account (which you should) you can see it/delete it at https://travis-ci.org/phac-nml/irida/caches.

I didn't think of having to remove the `node_modules` directory.  Is this only when you wish to test older code though?  That tends to be when I encounter this issue.  That is, would it ever come up if we are only ever testing newer and newer code?

I can remove it pretty easily though.

Reply:
I'd vote to remove the `node_modules` cache but keep the `m2` cache.  The node stuff doesn't take overly long to download fresh.

Reply:
Removed in f0735ef7a3564e10d4d064a43c1558ddc335adb0